common/models/user.js

```diff
@@ -558,6 +557,13 @@ module.exports = function(User) {
       return cb.promise;
     }
 
+    try {
+      if (options.password) {
+        UserModel.validatePassword(options.password);
```

Comment:
I don't think that's true. `resetMethod` does not touch the password, it only creates an access token. See also the API docs.

>```
    * Create a short lived acess token for temporary login. Allows users
    * to change passwords if forgotten.
    *
    * @options {Object} options
    * @prop {String} email The user's email address
    * @callback {Function} callback
    * @param {Error} err
    * @promise
```

Let's not worry about that right now and fix it as part of https://github.com/strongloop/loopback/issues/382.